internal/controller/upgradeinsights_controller.go

```diff
@@ -62,16 +53,19 @@ func (in *UpgradeInsightsController) Reconcile(ctx context.Context, req reconcil
 		}
 	}()
 
+	// Any condition changes prior to scope setup would not be persisted
+	utils.MarkCondition(ui.SetCondition, v1alpha1.ReadyConditionType, metav1.ConditionFalse, v1alpha1.ReadyConditionReason, "")
+	utils.MarkCondition(ui.SetCondition, v1alpha1.SynchronizedConditionType, metav1.ConditionFalse, v1alpha1.SynchronizedConditionReason, "")
+
 	// Handle resource deletion
-	result, err := in.handleDelete(ctx, ui)
+	result := in.handleDelete(ui)
 	if result != nil {
-		return *result, err
+		return *result, nil
 	}
 
 	// Sync UpgradeInsights with the Console API
 	err = in.sync(ctx, ui)
 	if err != nil {
-		logger.Error(err, "unable to save upgrade insights")
```

Comment:
will this break the poll loop since the controller reconcile is returning error?

Reply:
error has the priority over scheduled reconcile, so it will retry with the backoff limit until err is resolved and then get back to scheduled reconcile loop